test/test_va_api_get_create_config.cpp

```diff
@@ -54,19 +56,70 @@ TEST_P(VAAPIGetCreateConfig, CreateConfigWithAttributes)
         doQueryConfigEntrypoints(profile);
         if (doFindEntrypointInList(entrypoint)) {
             // profile and entrypoint are supported
-            doFillConfigAttribList();
-            doGetConfigAttributes(profile, entrypoint);
-            doCreateConfigWithAttrib(profile, entrypoint);
-            doDestroyConfig();
+            ConfigAttributes attribs;
+            getConfigAttributes(profile, entrypoint, attribs);
+
+            // create config with each individual supported attribute
+            for (const auto& attrib : attribs) {
+                const auto match = g_vaConfigAttribBitMasks.find(attrib.type);
+                if (match != g_vaConfigAttribBitMasks.end()) {
+                    // it's a bitfield attribute
+                    uint32_t bitfield(0);
+                    const BitMasks& masks = match->second;
+                    for (const auto mask : masks) { // for all bitmasks
+                        const bool isSet((attrib.value & mask) == mask);
+
+                        std::ostringstream oss;
+                        oss << std::hex << "0x" << attrib.type
+                            << ":0x" << attrib.value
+                            << ":0x" << mask << ":" << isSet;
+                        SCOPED_TRACE(oss.str());
+
+                        if (isSet) {
+                            // supported value
+                            bitfield |= mask;
+                            createConfig(profile, entrypoint,
+                                ConfigAttributes(
+                                    1, {type : attrib.type, value : mask }));
+                            destroyConfig();
+                        } else {
+                            // unsupported value
+                            const VAStatus expectation(
+                              (attrib.type == VAConfigAttribRTFormat) ?
+                                  VA_STATUS_ERROR_UNSUPPORTED_RT_FORMAT :
+                                  VA_STATUS_ERROR_INVALID_VALUE);
+                            createConfig(profile, entrypoint,
+                                ConfigAttributes(
+                                    1, {type : attrib.type, value : mask}),
+                                expectation);
+                            destroyConfig(VA_STATUS_ERROR_INVALID_CONFIG);
+                        }
+                    }
+                    // ensure we processed all supported values
+                    EXPECT_EQ(bitfield, attrib.value);
```

Comment:
VA_ENC_PACKED_HEADER_MISC is still used by some app, e.g. FFmpeg, so our driver still supports VA_ENC_PACKED_HEADER_MISC. however VA_ENC_PACKED_HEADER_MISC isn't included in gtest, so we can't use EXPECT_EQ for g_vaEncPackedHeaders.

Reply:
Yes, good catch.  I'll add VA_ENC_PACKED_HEADER_MISC to g_vaEncPackedHeaders